File: study_0728/src/study_0728/class03.java
package study_0728;

public class class03 {
	public static void main(String[] args) {
		

		int[] arr = new int[5];
		
//		arr[0] = 1;
//		arr[1] = 2;
//		arr[2] = 3;
//		arr[3] = 4;
//		arr[4] = 5;
		for(int i = 0; i<5; i++) {
			arr[i] = i+1;
		}
		
//		System.out.println(arr[0]);
//		System.out.println(arr[1]);
//		System.out.println(arr[2]);
//		System.out.println(arr[3]);
//		System.out.println(arr[4]);
		for(int i = 0; i<5; i++) {
			System.out.println(arr[i]);
		}
		
		
		// 400
		
		int[] arr2 = new int[400];
		
		for(int i = 0; i<400; i++) {
			arr2[i] = 2*i;
			System.out.print(arr2[i] + " ");
		}
		
		
		//
		System.out.println();
		//
		
		int[] arr3 = {1,2,3,4,5,6,7,8,9,6,8,9,0,1,2,3,4,5,6,3,4,8};
		for(int i=0; i<arr3.length; i++) {
			System.out.print(arr3[i] + " ");
		}
		
		
		
		
		
	}
}


// 배열(Array)
// 변수 -> 박스1
// 박스를 이어 놓은 것
// 자료형[] 배열명 = new 자료형[]

// 자료형 변수명; <- 선언 끝! 하지만 이대로 넘기면 쓰레기 값이 들어있음
// 자료형[] 변수명; <- 마찬가지로 쓰레기 값이 들어가는데 배열이라 너무 많은 쓰레기값이 들어간다

File: study_0824/Class03.py
# Numpy (Numerical Python)
# 과학, 공학 연산을 쉽게 하도록 지원하는 다차원 배열(multi-dimensional array) 라이브러리
# Numpy를 사용해서 대규모 배열 처리를 쉽게 하도록 하고, 파이썬 List 와는 차이가 있다.

# cmd -> pip numpy install

import numpy as np

a = np.array([1, 2, 3, 4, 5])
b = np.array([[1, 2, 3], [4, 5, 6]])

print(a)
print(b)

# 내부에 연속된 메모리 구조를 가지고 (Array Interface)를 가지고 있고, C를 통해 연산된다.

# 생성 함수 : np.array(), np.zeros(), np.ones(), np.empty(), np.arange(), np.linspace()
# 변환 함수 : ndarray.reshape(), ndarray.ravel(), ndarray.transpose(), ndarray.swapaxes()
# 연산 함수 : np.add(), np.substract(), np.multiply(), np.divide(), np.sqrt(), np.dot(), np.sum()
#                np.mean(), np.std(), np.max(), np.min(), np.argmax(), np.argmin()
# 집계 함수: ndarray.sum(), ndarray.mean(), ndarrray.std(), ndarray.max(), ndarray.min(),
#               ndarray.argmax(), ndarray.argmin()
# 논리함수 : np.logical_and(), np.logical_or(), np.logical_not()


# 생성 함수
arr1 = np.zeros(5)
print(arr1)

arr2 = np.zeros((2,8))
print(arr2)

arr3 = np.zeros((3,3,3))
print(arr3)

arr1 = np.ones(5)
print(arr1)
arr2 = np.ones((10,10))
print(arr2)

arr4 = np.empty(5)
print(arr1)
arr5 = np.empty((3,3))
print(arr5)

arr1 = np.arange(5)
print(arr1)
arr2 = np.arange(3,10)
print(arr2)
arr3 = np.arange(1, 10, 0.2)
print(arr3)

arr1 = np.linspace(0,1,5)
print(arr1)
arr2 = np.linspace(-10, 10, 10)
print(arr2)

# 변환 함수
# 배열 형태 변환

# 1차원 배열을 2차원 배열도 변환
arr = np.array([1,2,3,4,5,6])
arr2 = arr.reshape(2,3)
print(arr2)

# 2차원 배열을 1차원 배열로 변환
arr1 = arr2.flatten()
print(arr1)

# 타입 변환

# 정수를 실수로 변환
arr_int = np.array([1,2,3])
arr_float = arr_int.astype(float)
print(arr_float)

# 문자형 배열을 정수형 배열로 변환
arr_str = np.array(['1', '2', '3'])
arr_int = arr_str.astype(int)
print(arr_int)

# 축 변환
